### ISSUE TITLE
spectra from mobility mzml files cannot be extracted when the experiments involve selection/fragmentation

### ISSUE THREAD

**Issue by lopippo** [OpenMS/OpenMS#1850]
Greetings, 

I am having trouble iterating into an OnDiscMSExperiment<> instance when that is instance
loads an mzML file that contains mobility data (Waters Synapt 2) from an experiment that involves ion selection/fragmentation either before or after the mobility separation. Interestingly, I can get the number of spectra fine, even if the iteration in  then fails. (See code below)

Mobility experiments can be loaded and parsed fine as long as they do not involve precursor ions selection. 
```
 // Load data from an indexed MzML file
 OnDiscMSExperiment<> ondmseMap;
 imzml.load(globalOptions.mzmlFilename(), ondmseMap);

 QString dateTimeString = expSettings->getDateTime().getDate().toQString();
(works fine)

 /* the number of spectra in this whole mzml file */
 uint numberOfSpectra = ondmseMap.getNrSpectra();
 specMetaData.setIntItem("spectrumListCount", numberOfSpectra);
(works fine)

 for(uint specIndex = 0; specIndex < numberOfSpectra; ++specIndex)
 {
  MSSpectrum<> spectrum = ondmseMap.getSpectrum(specIndex);
(crashes)
```

OpenMS version 2.0.0 gotten from GitHub dated 20160120.

How can I help ? I could provide the three files where I see the crash:

- simple quadrupole selection -> mobility -> time-of-flight
- quadrupole selection -> fragmentation -> mobility -> time-of-flight
- quadrupole selection -> mobility -> fragmentation -> time-of-flight

Only roughly 6-7 Mb each.

Sincerely,
Filippo Rusconi
Debian packager of OpenMS (but here with my scientist hat on)

**Comment by timosachsenberg** [comment]
Hi Filippo. It would be great if you could provide the mzML files e.g. as github gist.

**Comment by hroest** [comment]
the problem may lie with the mzml files that contain empty `<binary></binary>` XML tags. I hope that #1860  will fix the issue

@lopippo can you try the branch hroest fix/mzml_decoder

**Comment by timosachsenberg** [comment]
any update here? the PR seemed rather complete but travis fall failing

**Comment by hroest** [comment]
fixed by #1860
